Src/AutoFixture/ElementsBuilder.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Ploeh.AutoFixture.Kernel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Draws a random element from the given collection
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public sealed class ElementsBuilder<T> : ISpecimenBuilder
+    {
+        private readonly ICollection<T> elements;
+        private readonly RandomNumericSequenceGenerator sequence;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ElementsBuilder{T}"/> class.
+        /// </summary>
+        /// <param name="elements">The collection from which elements should be drawn from.
+        /// It must contain at least two element.</param>
+        public ElementsBuilder(ICollection<T> elements)
+        {
+            if (elements == null)
+                throw new ArgumentNullException("elements");
+
+            if (elements.Count < 2)
```

Comment:
I was struggling with that as well; my initial unit tests were using a single element collection.

My idea was to internally use ``RandomNumericSequenceGenerator`` in order to return elements from the collection in a random order but still returning each element once before resetting.

Because of the use of ``RandomNumericSequenceGenerator`` which constructor needs at least an upper and a lower limit that are strictly ascending (see #408 and #409), these were my options:
* Make special case for one element collections, which is a bit ugly
* Force the collection to have at least 2 elements to have a simple implementation
* Stop using ``RandomNumericSequenceGenerator``, for example by using ``NumericSequenceGenerator`` instead but then element would always be returned in the same order - I kind of liked the idea that element would be returned at random :smile: 
* Use ``Random`` and give up on the idea of returning each element once

Any advice welcome, I made the compromise based solely on personal preferences